### ISSUE TITLE
Include output from jailer.sh to log if it fails

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#33175]
**What kind of request is this (question/bug/enhancement/feature request):** Enhancement

**Other details that may be helpful:**
In case `jailer.sh` script fails during Rancher startup, script's output is not included in the log, only FATAL message like `[FATAL] error running the jail command: exit status 1` is produced making it hard to pinpoint what exactly caused this script to fail.
Currently `jailer.sh` script output is added at DEBUG level only when if it succeeds.
It's suggested to log script output in all cases and also switch it to TRACE level to prevent any sensitive information from appearing in logs (sensitive information is acceptable at TRACE level).
Note it's currently not trivial to observe current behavior as `jailer.sh` doesn't fail. It was discovered as part of working on https://github.com/rancher/rancher/issues/33127.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head (cdc12d8a5cc6fda069709a950f9a6dc9e38892e1).

**Comment by snasovich** [comment]
This is now on `master-head`.
@sowmyav27 , what is our approach on testing issues that are not reproducible unless code is modified for it to fail? The only option I see is to produce a custom build with modified `jailer.sh` that fails so when this version is deployed you would see script output is logged. This does sound like overkill for something with such trivial fix (see linked PR).

**Comment by sowmyav27** [comment]
Waiting for info from @snasovich to provide info on how to reproduce/test this

**Comment by snasovich** [comment]
@kravciak , the idea for testing this is to replace `jailer.sh` file with one that fails before running Rancher.

Here is a script that can be used to test it:

```sh
#!/bin/bash
set -e

IMAGE=$1

if [[ $IMAGE = "" ]]
  then
    echo "One argument (image name with tag) must be passed to the script"
    exit 1
fi

CONTAINER_ID=$(docker create --restart=unless-stopped --privileged --pull=always $IMAGE --trace=true)

printf "#!/bin/bash\n\ncp /usr/path-doesnt-exist /usr" > jailer.sh
docker cp jailer.sh $CONTAINER_ID:/usr/bin/jailer.sh

docker start $CONTAINER_ID

docker logs $CONTAINER_ID -f

docker rm $CONTAINER_ID -f
```

If running it passing `rancher/rancher:master-head`, you can see there is TRACE log message with `jailer.sh` script output that allows understanding how exactly the script has failed.

If running it with version prior to the fix (e.g. `rancher/rancher:master-54d4afc27c36814fe067a47a232463e0e5681da9-head`), there is no TRACE log message with script output, just FATAL one stating that sript failed with exit status 1.

Note that if running it on `rancher/rancher:v2.5.8`, the container starts OK - seems failures in `jailer.sh` previously did not prevent Rancher from being started.

**Comment by superseb** [comment]
Another way to test this is running with a read-only mounted `/opt/jail`:

```
docker run --privileged -v /tmp/jail:/opt/jail:ro rancher/rancher:master-head --trace
```

**Comment by kravciak** [comment]
I reproduced on 2.5.8 and checked the fix on master-head (3dae3e2). Startup fails with error in docker logs:
```
2021/06/17 14:28:45 [INFO] Done waiting for CRD templates.management.cattle.io to become available
2021/06/17 14:28:48 [FATAL] error running the jail command: exit status 1
```
